src/components/SwapForm/service/getBalanceNear.ts

```diff
@@ -0,0 +1,35 @@
+import { formatUnits } from "viem"
+
+import { nearAccount } from "@src/utils/near"
+import {
+  NEAR_MIN_RESERVE_BALANCE,
+  NEAR_STORAGE_COST_PRE_BYTE,
+} from "@src/constants/contracts"
+
+export const minimumNearBalance = (storageUsed: number): number => {
+  const storageCostPerByte = NEAR_STORAGE_COST_PRE_BYTE
+  const calculateMinBalance = storageUsed * storageCostPerByte
+  return storageUsed <= 770
+    ? 0
+    : calculateMinBalance > NEAR_MIN_RESERVE_BALANCE
+      ? calculateMinBalance
+      : NEAR_MIN_RESERVE_BALANCE
+}
+
+export const getBalanceNear = async (accountId: string) => {
+  try {
+    const viewAccount = await nearAccount(accountId)
+    if (!viewAccount) {
+      return 0
+    }
+    const formattedAmountOut = formatUnits(BigInt(viewAccount.amount), 24)
```

Comment:
It is way better to make functions that interact with NEAR to return  just strings (or rawest format possible) and do all the formatting outside.

Reply:
I don't want a separate functions, just pushing for working with BigNumber when possible and do all the formatting outside.

P.S. You can leave it like this if you want

Reply:
Why do we need here BigNumber?